### DISCUSSION TITLE
How can i calculate "scale"?

### DISCUSSION THREAD

**Discussion by antpv** [dinerojs/dinero.js#446] [Q&A] [answered]
I have currency rate like this: 0.0001234

How can i calculate "scale" and "amount" for dinero currency rate correctly?

**Comment by sarahdayan** [comment, marked as answer]
Hey @antpv!

Your rate has seven fraction digits, meaning you can represent as an integer with a scale of `7`.

An easy way to get the `amount` is to then multiply your rate by 10 to the power of the scale (`0.0001234 * 10**7`), so in your case it would be `{ amount: 1234, scale: 7 }`.

**Reply by antpv** [reply]
Thanks @sarahdayan!

What about converting currencies with different exponent and base values?

For example, there is a currency:
`{ code: 'MRU', base: 5, exponent: 1 }`

I want to convert it to currency:
`{ code: 'UAH', base: 10, exponent: 2 }`

I have a course (UAH to MRU) from exchange rate service 0.727655

Using your function I generated rate
```
{
  UAH: { amount: 727655, scale: 6 }
}
```

But when converting i get 7.27 UAH, instead of 0.27 UAH

**Reply by sarahdayan** [reply]
There seems to be an issue when converting between currencies with different bases, thank you for catching it @antpv! I'm creating an issue now and will look into it.

**Reply by sarahdayan** [reply]
Issue open: https://github.com/dinerojs/dinero.js/issues/477

If you're interested in picking it up @antpv feel free!
